modules/i3s/src/types.ts

```diff
@@ -311,24 +367,85 @@ type Store = {
   lodModel: string;
   defaultGeometrySchema: DefaultGeometrySchema;
 };
-
+// Spec - https://github.com/Esri/i3s-spec/blob/master/docs/1.8/defaultGeometrySchema.cmn.md
 type DefaultGeometrySchema = {
-  vartexCount: number;
-  featureCount: number;
-  position: Float32Array;
-  normal: Float32Array;
-  uv0: Float32Array;
-  color: Uint8Array;
-  id: Float32Array;
-  faceRange: Uint32Array;
-  region: Uint16Array;
-};
-
-// TODO change string to possible values from https://github.com/Esri/i3s-spec/blob/master/docs/1.8/heightModelInfo.cmn.md
+  geometryType?: 'triangles';
+  topology: 'PerAttributeArray' | 'Indexed';
+  header: {
+    property: 'vertexCount' | 'featureCount' | string;
+    type:
+      | DTYPE_LOOKUP.UInt8
+      | DTYPE_LOOKUP.UInt16
+      | DTYPE_LOOKUP.UInt32
+      | DTYPE_LOOKUP.UInt64
+      | DTYPE_LOOKUP.Int16
+      | DTYPE_LOOKUP.Int32
+      | DTYPE_LOOKUP.Int64
+      | DTYPE_LOOKUP.Float32
+      | DTYPE_LOOKUP.Float64;
+  }[];
+  ordering: string[];
+  vertexAttributes: vertexAttribute;
+  faces?: vertexAttribute;
+  featureAttributeOrder: string[];
+  featureAttributes: FeatureAttribute;
+  // TODO Do we realy need this Property?
+  attributesOrder?: string[];
+};
+export type vertexAttribute = {
+  position: geometryAttribute;
+  normal: geometryAttribute;
+  uv0: geometryAttribute;
+  color: geometryAttribute;
+  region?: geometryAttribute;
+};
+export type geometryAttribute = {
+  byteOffset?: number;
+  valueType:
+    | DTYPE_LOOKUP.UInt8
+    | DTYPE_LOOKUP.UInt16
+    | DTYPE_LOOKUP.Int16
+    | DTYPE_LOOKUP.Int32
+    | DTYPE_LOOKUP.Int64
+    | DTYPE_LOOKUP.Float32
+    | DTYPE_LOOKUP.Float64;
+  valuesPerElement: number;
+};
+export type normalizedAttributes = {
+  [key: string]: normalizedAttribute;
+};
+export type normalizedAttribute = {
+  value: number[] | TypedArray;
+  type: number;
+  size: number;
+  normalized?: boolean;
+  metadata?: any;
+};
 type HeightModelInfo = {
-  heightModel: string;
+  heightModel: 'gravity_related_height' | 'ellipsoidal';
   vertCRS: string;
-  heightUnit: string;
+  heightUnit:
```

Comment:
We have a `HEIGHT_MODEL_INFO` object with the following values:

> const HEIGHT_MODEL_INFO = {
  heightModel: {
    path: 'heightModel',
    default: 'gravity_related_height'
  },
  vertCRS: {
    path: 'vertCRS',
    default: 'EGM96_Geoid'
  },
  heightUnit: {
    path: 'heightUnit',
    default: 'meter'
  }
};
And we don't change it while 3DTiles -> I3S conversion process.

Reply:
- I don't think 3D tiles format supports those height models, so preserving it doesn't really make sense?
- The `Tile3DLayer` does not support those formats

So my expectation is that an I3S tileset with non-standard heightUnit will neither convert nor display correctly.
- I suspect those tilesets are very rare, so it might be enough to issue a warning.
- I also suspect that supporting these would be fairly simple as I assume we just need to apply a scale factor on height, could perhaps even be done as part of modelMatrix...